lib/chef/resource/kernel_module.rb

```diff
@@ -0,0 +1,100 @@
+#
+# Resource:: kernel_module
+#
+# The MIT License (MIT)
+#
+# Copyright 2016-2018, Shopify Inc.
+# Copyright 22018, Chef Software, Inc.
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class KernelModule < Chef::Resource
+      resource_name :kernel_module
+      provides(:kernel_module) { true }
+
+      property :modname, String, name_property: true, identity: true
+      property :load_dir, String, default: "/etc/modules-load.d"
+      property :unload_dir, String, default: "/etc/modprobe.d"
+
+      # Load kernel module, and ensure it loads on reboot
+      action :install do
+        declare_resource(:directory, new_resource.load_dir) do
+          recursive true
+        end
+
+        declare_resource(:file, "#{new_resource.load_dir}/#{new_resource.modname}.conf") do
+          content "#{new_resource.modname}\n"
+          notifies :run, "execute[update initramfs]"
+        end
+
+        declare_resource(:execute, "update initramfs") do
+          command initramfs_command
+          action :nothing
+        end
+
+        new_resource.run_action(:load)
```

Comment:
I'm not sure this does what you want. This would run the entire load action before the install action's resources run.

Reply:
That's the goal.